generator: local/remote commit comment bug?

I was verifying #869 and getting some weird results. 

When I gen using the remote this I get in the header: 

```go
// Source: exercism/problem-specifications
// Commit: 196fc1a anagram: Rename duplicated test case description (#671)

There are some tracks that generate their test suits and use the description field as name for the test function/method.

* Bump version number
// Problem Specifications Version: 1.0.1
```
which ends up being a syntax error. The extended comment lines are not commented out. Generator confirms this:

```text
$ go run exercises/anagram/.meta/gen.go
2017/09/30 15:24:51 [LOCAL] fetching anagram test data
2017/09/30 15:24:51 [LOCAL] No test data found
2017/09/30 15:24:51 [REMOTE] fetching anagram test data
2017/09/30 15:24:53 [REMOTE] source: https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/anagram/canonical-data.json
2017/09/30 15:24:53 [ERROR] format.Source failed. The generated source has a syntax error.
2017/09/30 15:24:53 [ERROR] output: /home/tml/projects/exercism-development/github.com/shaleh/exercism-go/exercises/anagram/cases_test.go
2017/09/30 15:24:53 8:1: expected declaration, found 'IDENT' There
exit status 1
```

When I run against the generator with a current local filesystem repo I get this:

```go
// Source: exercism/problem-specifications
// Commit: 196fc1a anagram: Rename duplicated test case description (#671)
// Problem Specifications Version: 1.0.1
```

So I think when it pulls remote it pulls the [entire commit message](https://github.com/exercism/problem-specifications/commit/196fc1a2e8274ee082c35442ea4e8b9c8872d9c1) not just the first line and drops it into the template. Which if uncommented will cause a syntax error for Go.

I'm looking into this, but wanted to note it in case anyone sees the issue right away, and I want to ref the bug in #869.

--- a/gen/gen.go
+++ b/gen/gen.go
@@ -13,6 +13,7 @@ import (
 	"os/exec"
 	"path/filepath"
 	"runtime"
+	"strings"
 	"text/template"
 	"time"
 )
@@ -242,5 +243,7 @@ func getRemoteCommit(exercise string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return fmt.Sprintf("%s %s", c[0].Sha[0:7], c[0].Commit.Message), nil
+	// Use only 1st line of the commit message
+	lines := strings.SplitN(c[0].Commit.Message, "\n", 2)
+	return fmt.Sprintf("%s %s", c[0].Sha[0:7], lines[0]), nil
 }

